### ISSUE TITLE
release 1.8: external libraries

### ISSUE THREAD

**Issue by ousia** [rk700/PyMuPDF#29]
I’m not a programmer myself, but this is the first time I see a binding library that contains already compiled third-party binaries.

mupdf itself contains a directory named `thirdparty`, but only containing source code (Fedora removes this directory before compilation).

I will ask to Fedora developers to add this library to repositories for the current development version. Would it be possible that both `LibLinux` and `LibWin32` directories are moved to a different branch from the default master branch?

In principle, any Linux distribution should compile any binary using the already installed libraries and not using binaries provided with the source code to be compiled.

Maybe in Windows this is harder to achieve. But in any case, would it be possible to add those binaries to another branch (such as `external-libraries`)?

Sorry for being so assertive, but I guess this is mandatory for many distributions to include PyMuPDF in their repositories.

**Comment by rk700** [comment]
@ousia it does make sense for linux. Currently we need libmupdf and libmujs for linking, and it would be fine as long as the these libs are provided by the linux distribution.

IMHO, we can remove the libs in PyMuPDF and put them in a separate repo. Then by using `git submodule` https://git-scm.com/book/en/v2/Git-Tools-Submodules, users can choose either the libs provided by the system or from us.

@JorjMcKie, what do you think?

**Comment by JorjMcKie** [comment]
I do agree with both of you. Users should be able to choose how they want to install PyMuPDF, and thus be able to de-slect downloading parts of what we provide.
I am a Windows user. When I tried to install MuPDF 1.2 for python-fitz (months ago now), I had to install MS Visual Studio first, which is a major effort if you just want to use MuPDF in Python. Providing lib modules with PyMuPDF abolishes this need and therefor makes much sense.

**Comment by rk700** [comment]
All right, so let's break down the steps:
1. I suggest that we only provide 3rd party libs for windows users, and let linux users use the system's libs.
2. @JorjMcKie could you please create a new repo which contains those windows libs? We will use that repo as a submodule of PyMuPDF
3. After step 2 is finished, we update the setup.py and README.

**Comment by JorjMcKie** [comment]
@rk700 Done. This the link:
https://github.com/JorjMcKie/PyMuPDF-optional-material
I have included / copied the Winlib32 directory to it, so it can be deleted here.
I also copied the WIndows help file ``PyMuPDF.chm`` to that repository. It is another example of a file only of interest to Windows users. This file can therefor also be deleted in this rep.

**Comment by ousia** [comment]
Many thanks for the changes, @rk700 and @JorjMcKie.

Before the release 1.8, only the directories are to be removed and the submodule added.

**Comment by ousia** [comment]
I submitted the changes myself in a merge request to the 1.8 branch.

**Comment by rk700** [comment]
@JorjMcKie Thank you for the work!

I've merged the PR from @ousia and updated the setup.py. @JorjMcKie could please help again and see whether the compilation works on windows?

The 1.8 branch will be merged into master when all are done.

**Comment by JorjMcKie** [comment]
Will do that. Pls give me a few days - I am having issues with my internet connection ...

**Comment by JorjMcKie** [comment]
OK - I have tested the compilation with setup.py of the 1.8 branch. 
All worked fine.
@rk700: After the merge (you?), I will do a final check to update documentation, etc. as needed.

**Comment by JorjMcKie** [comment]
I have updated documentation so that it now reflects the changed repository structure (i.e. includes optional material).
@rk700  - close issue?

**Comment by ousia** [comment]
Will we have a proper 1.8 release or are there other things that come first?

**Comment by rk700** [comment]
@JorjMcKie filed a bug https://github.com/rk700/PyMuPDF/issues/32. I guess it'd be okay to create the 1.8 release when the fix is verified.

**Comment by ousia** [comment]
Since issue #32 has been fixed, how about releasing version 1.8 of PyMuPDF?

**Comment by rk700** [comment]
Right I guess that there're no fatal bugs now and version 1.8 is released: https://github.com/rk700/PyMuPDF/releases/tag/v1.8

**Comment by ousia** [comment]
Many thanks for the new release and the excellent work, @rk700, @JorjMcKie and @deepgully.

**Comment by rk700** [comment]
Here's the PKGBUILD of archlinux for your reference: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=python2-pymupdf

Not quite familiar with rpm/deb, but basically should be the same.
